docs/testing/workspaces-migration-guide.md

```diff
@@ -0,0 +1,319 @@
+---
+sidebar_position: 3
+sidebar_label: Workspaces Migration Guide
+title: "Workspaces Migration Guide"
+---
+
+# Workspaces Migration Guide
+
+NEAR Simulator was meant to be an in-place replacement of a blockchain environment for the purpose of testing NEAR contracts. However, simulating NEAR ledger turned out to be a much more complex endeavour than was anticipated. Eventually, the idea of workspaces was born - a library for automating workflows and writing tests for NEAR smart contracts using a real NEAR network (localnet, testnet or mainnet). Thus, NEAR Simulator is being deprecated in favor of [`workspaces-rs`](https://github.com/near/workspaces-rs), the Rust edition of workspaces. As the two libraries have two vastly different APIs this guide was created to ease the migration process for developers.
+
+:::danger
+TODO: I do not have a whole lot of context here why exactly simtests were not suitable for our purposes, so if anyone wants to elaborate the preceding paragraph please do.
```

Comment:
Can someone help filling this gap? @ChaoticTempest you probably have the most amount of context here, feel free to write it as a comment or push a commit to my branch.

Reply:
A few reasons I can remember:
- near-sdk-sim was hooking into parts of nearcore that were not meant to be released, in the most recent version those crates aren't released so sim is currently using duplicate deps (maintenance nightmare)
- Not a fully accurate simulation because it just used a subset of the runtime in a specific way -- can't rely on this
- Bulky dependencies pulled in (drastically increases compile time)
- Unergonomic API, not specific to this strategy, but likely would have had to be re-built
- Can't test parallel transactions easily -- current pattern would process blocks until a transaction succeeded but you can't create specific conditions, which is required for a strategy like this that isn't fully simulated

I'll let Phuong share opinions when he's back since he likely will have some inputs for this

Reply:
> Not a fully accurate simulation because it just used a subset of the runtime in a specific way -- can't rely on this

And thus couldn't measure gas burnt accurately. Also, all the intricacies of nearcore (like protocol features) wouldn't be one-to-one with the runtime since the runtime was just code built on top of VM logic. 

People would also need to write their own automation scripts to deploy to testnet, so we'd end up with very split workflows for testing.